qpc/insights/upload.py

```diff
@@ -147,13 +169,51 @@ def _validate_args(self):
                      self.report_id)))
         print(_(messages.INSIGHTS_RETRIEVE_REPORT % self.report_id))
 
+    def verify_report_details(self):
+        """
+        Verify that the report contents are a valid deployments report.
+
+        :returns boolean regarding report validity
+        """
+        # validate required keys
+        report_contents = extract_json_from_tar(self.response.content,
+                                                print_pretty=False)
+        required_keys = ['report_platform_id',
+                         'report_id',
+                         'report_version',
+                         'report_type',
+                         'system_fingerprints']
+        for key in required_keys:
+            present = report_contents.get(key)
+            if not present:
+                print(_(messages.INSIGHTS_REPORT_MISSING_FIELD % key))
+                return False
+        report_id = report_contents.get('report_id')
+        # validate report type
+        report_type = report_contents.get('report_type')
+        if report_type != 'deployments':
+            print(_(messages.INSIGHTS_INVALID_REPORT_TYPE % report_id))
+            return False
+
+        # validate fingerprints contain canonical facts
+        fingerprints = report_contents.get('system_fingerprints')
+        verified_fingerprints = verify_report_fingerprints(fingerprints, report_id)
+        if not verified_fingerprints:
+            print(_(messages.INSIGHTS_REPORT_NO_VALID_FP % report_id))
```

Comment:
If there are some invalid fingerprints, we should still send.  This log message should be changed.  If there are no valid fingerprints, then the report is invalid.